src/squlearn/encoding_circuit/circuit_library/chebyshev_pqc.py

```diff
@@ -143,8 +149,14 @@ def generate_initial_parameters(self, seed: Union[int, None] = None) -> np.ndarr
     def feature_bounds(self) -> np.ndarray:
         """The bounds of the features of the ChebyshevPQC encoding circuit."""
         bounds = np.zeros((self.num_features, 2))
-        bounds[:, 0] = -1.0
-        bounds[:, 1] = 1.0
+        if self.phi_map == "arccos":
+            bounds[:, 0] = -1.0
+            bounds[:, 1] = 1.0
+        elif self.phi_map == "arctan":
+            bounds[:, 0] = -np.inf
+            bounds[:, 1] = np.inf
+        else:
+            raise ValueError("Unknown value for phi_map: ", self.phi_map)
```

Comment:
How can this case occur? Only through editing after initialization? Shouldn't we make phi_map protected instead of checking here?

Reply:
parameters that occur in the signature of `__init__` can be changed via `set_params`. But maybe we could move this check to `set_params`

Reply:
that's what i meant, making it private and then moving all checks in a dedicated setter

Reply:
alright, i'll move it 🙂